### PR TITLE
FS-82 Add file upload api

### PR DESCRIPTION
**Pull request by mic-smith**
## Description

Add `/uploadfile` endpoint for uploading files (txt or pdf). PDF files have the text extracted using pypdf.
Text from the files is stored in redis.

## Changelog
- Add `/uploadfile` POST endpoint for uploading  files
- GET endpoint added for testing
- Added default redis.conf and docker file to bundle. Disabled persistence in redis

### CODE REVIEW

File: backend/src/file_upload_service.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        pdf_file = PdfReader(file.file)
+        all_content = ""
+        for page_num in range(len(pdf_file.pages)):
+            page_text = pdf_file.pages[page_num].extract_text()
+            all_content += page_text
+            all_content += "\n"
+
+        # pdfminer.six processing
+        # out_str = StringIO()
+        # extract_text_to_fp(file.file, out_str) #, codec='cp1251', strip_control=True)
```
Comment (IMladjenovic) [inline backend/src/file_upload_service.py:33]: Do we need this? If so can we put a comment explaining why we've left this in?

File: backend/src/file_upload_service.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        # all_content = out_str.getvalue()
+
+        end_time = time.time()
+
+        logger.debug(f'PDF content {all_content}')
+        logger.info(f"PDF content extracted successfully in {(end_time - start_time)}")
```
Comment (IMladjenovic) [inline backend/src/file_upload_service.py:39]: 👍

File: backend/src/session/file_uploads.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+
+def clear_session_file_uploads_meta():
+    logger.info("File Uploads session cleared")
+
+    meta_list = get_session(UPLOADS_META_SESSION_KEY, [])
+
+    keys = []
+    for meta in meta_list:
+        keys.append(UPLOADS_KEY_PREFIX + meta["uploadId"])
```
Comment (IMladjenovic) [inline backend/src/session/file_uploads.py:61]:
```suggestion
    keys = [ UPLOADS_KEY_PREFIX + meta["uploadId"] for meta in meta_list ]
```
I think list comprehension can tidy this up slightly

File: redis/redis.conf [hunk 4]
```diff
@@ -0,0 +1,2326 @@
+# Redis configuration file example.
```
Comment (IMladjenovic) [inline redis/redis.conf:1]: Do we need this whole config file, or can we just overwrite the default configuration somehow?

**Review by gaganahluwalia — APPROVED** [review]
Approved.

File: backend/src/file_upload_service.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+from pypdf import PdfReader
+from src.session.file_uploads import FileUpload, update_session_file_uploads, get_session_file_upload
+
+logger = logging.getLogger(__name__)
+
+MAX_FILE_SIZE = 10*1024*1024
+
+def handle_file_upload(file:UploadFile) -> str:
+
+    if (file.size or 0) > MAX_FILE_SIZE:
```
[the remark this answers is not in the document]
Comment (mic-smith) [reply backend/src/file_upload_service.py:16]: No, it was always set in test. I've updated the local FileUpload type to match with the None to simplify copying values below.

File: backend/src/file_upload_service.py (hunk 1 above)
Comment (mic-smith) [reply backend/src/file_upload_service.py]: Removed

File: backend/src/session/file_uploads.py (hunk 3 above)
Comment (mic-smith) [reply backend/src/session/file_uploads.py]: Nice, thanks

File: redis/redis.conf (hunk 4 above)
Comment (mic-smith) [reply redis/redis.conf]: Seems to work as just config file with the single no save prop override. I've updated

**Review by IMladjenovic — APPROVED** [review]
Approved. If you can add a link to where you found the default redis config file online that would be helpful if we need to make changes in the future. Minor detail though

**Pull request merged**